### PR TITLE
Add vsphere-pv-csi 2.4.0 package

### PR DESCRIPTION
**Pull request by shyaamsn**
## What this PR does / why we need it
Add vsphere-pv-csi 2.4.0 package

## Describe testing done for PR
ytt rendering and checked the manifest is as expected

## Special notes for your reviewer

### CODE REVIEW

File: addons/packages/vsphere-pv-csi/2.4.0/bundle/config/overlays/set-cns-csi-conf.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#@ load("@ytt:overlay", "overlay")
+#@ load("/values.star", "values")
+
+#@overlay/match by=overlay.subset({"kind": "ConfigMap", "metadata": {"name": "pvcsi-config"}}),expects=1
+---
+data:
+  #@overlay/replace
+  cns-csi.conf: #@ "[GC]" + "\nendpoint = %s\nport = %d\ntanzukubernetescluster-uid = %s\ntanzukubernetescluster-name = %s\ncluster-distribution = \"TKGService\""%(values.vspherePVCSI.supervisor_master_endpoint_hostname,values.vspherePVCSI.supervisor_master_port,values.vspherePVCSI.cluster_uid,values.vspherePVCSI.cluster_name)
```
Comment (blc1996) [inline addons/packages/vsphere-pv-csi/2.4.0/bundle/config/overlays/set-cns-csi-conf.yaml:8]: This line is a bit different from the template of 2.4.1, please confirm it's expected.

```
2.4.1:
cns-csi.conf: #@ "[GC]" + "\nendpoint = %s\nport = %d\ntanzukubernetescluster-uid = %s\ntanzukubernetescluster-name = %s"%(values.vspherePVCSI.supervisor
_master_endpoint_hostname,values.vspherePVCSI.supervisor_master_port,values.vspherePVCSI.cluster_uid,values.vspherePVCSI.cluster_name)
```

File: addons/packages/vsphere-pv-csi/2.4.0/bundle/.imgpkg/images.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    kbld.carvel.dev/id: gcr.io/cloud-provider-vsphere/csi/release/syncer:v2.4.0
+  image: gcr.io/cloud-provider-vsphere/csi/release/syncer@sha256:b6da4448adf8cc2eb363198748d9f26d17acb500a80c53629379d83cecb6cefe
+- annotations:
+    kbld.carvel.dev/id: k8s.gcr.io/sig-storage/csi-attacher:v3.3.0
+  image: k8s.gcr.io/sig-storage/csi-attacher@sha256:80dec81b679a733fda448be92a2331150d99095947d04003ecff3dbd7f2a476a
+- annotations:
+    kbld.carvel.dev/id: k8s.gcr.io/sig-storage/csi-node-driver-registrar:v2.3.0
+  image: k8s.gcr.io/sig-storage/csi-node-driver-registrar@sha256:f9bcee63734b7b01555ee8fc8fb01ac2922478b2c8934bf8d468dd2916edc405
+- annotations:
+    kbld.carvel.dev/id: k8s.gcr.io/sig-storage/csi-provisioner:v3.0.0
```
Comment (blc1996) [inline addons/packages/vsphere-pv-csi/2.4.0/bundle/.imgpkg/images.yml:17]: Could you use a newer version of kbld to rerun the lock, tag information like following are dropped

```
-    kbld.carvel.dev/origins: |
-      - resolved:
-          tag: v3.0.0
-          url: k8s.gcr.io/sig-storage/csi-provisioner:v3.0.0
```

File: addons/packages/vsphere-pv-csi/2.4.0/bundle/config/overlays/set-cns-csi-conf.yaml (hunk 1 above)
Comment (shyaamsn) [reply addons/packages/vsphere-pv-csi/2.4.0/bundle/config/overlays/set-cns-csi-conf.yaml:8]: Yes thats intentional. cluster-distribution = "TKGService" is needed.

File: addons/packages/vsphere-pv-csi/2.4.0/bundle/.imgpkg/images.yml (hunk 2 above)
Comment (shyaamsn) [reply addons/packages/vsphere-pv-csi/2.4.0/bundle/.imgpkg/images.yml:29]: Updated

**Comment by shyaamsn** [comment]
Not required anymore since vsphere-pv-csi will not be a upstream package anymore since it is only applicable to vSphere Tanzu supervisor cluster

**Pull request closed without merging**